fuel-tests/tests/contract.rs

```diff
@@ -1,29 +1,201 @@
+use fuel_tx::TransactionBuilder;
+use fuel_vm::{consts::*, prelude::*};
+use itertools::Itertools;
+use rand::{rngs::StdRng, Rng, SeedableRng};
 
 #[tokio::test]
-async fn contract() {
-    // setup test data in the node
-    let contract = FuelVmContract::default();
-    let id = fuel_types::ContractId::new(Default::default());
-
-    let mut db = Database::default();
-    Storage::<fuel_types::ContractId, FuelVmContract>::insert(&mut db, &id, &contract).unwrap();
-    // setup server & client
-    let srv = FuelService::from_database(db, Config::local_node())
+async fn test_contract_salt() {
+    let mut rng = StdRng::seed_from_u64(2322);
+    let mut test_builder = TestSetupBuilder::new(2322);
+    let (_, contract_id) = test_builder.setup_contract(vec![]);
+
+    // initialize 10 random transactions that transfer coins and call a contract
+    let transactions = (1..=10)
+        .into_iter()
+        .map(|i| {
+            let secret = SecretKey::random(&mut rng);
+            TransactionBuilder::script(
+                Opcode::RET(REG_ONE).to_bytes().into_iter().collect(),
+                vec![],
+            )
+            .gas_limit(100)
+            .gas_price(1)
+            .byte_price(1)
+            .add_unsigned_coin_input(
+                rng.gen(),
+                &secret,
+                1000 + i,
+                Default::default(),
+                0,
+                vec![],
+                vec![],
+            )
+            .add_input(Input::Contract {
+                utxo_id: Default::default(),
+                balance_root: Default::default(),
+                state_root: Default::default(),
+                contract_id,
+            })
+            .add_output(Output::Change {
+                amount: 0,
+                asset_id: Default::default(),
+                to: rng.gen(),
+            })
+            .add_output(Output::Contract {
+                input_index: 1,
+                balance_root: Default::default(),
+                state_root: Default::default(),
+            })
+            .finalize()
+        })
+        .collect_vec();
+
+    // setup genesis block with coins that transactions can spend
+    test_builder.config_coin_inputs_from_transactions(&transactions);
+
+    // spin up node
+    let TestContext { client, .. } = test_builder.finalize().await;
+
+    let contract = client
+        .contract(format!("{:#x}", contract_id).as_str())
         .await
         .unwrap();
-    let client = FuelClient::from(srv.bound_address);
 
-    // run test
-    let contract = client
-        .contract(format!("{:#x}", id).as_str())
+    // Check that salt is 0x Hex prefixed
+    let salt = contract.unwrap().salt;
+    assert_eq!("0x", &salt.to_string()[..2]);
+}
+
+#[tokio::test]
+async fn test_contract_balance_default() {
+    let mut rng = StdRng::seed_from_u64(2322);
+    let mut test_builder = TestSetupBuilder::new(2322);
+    let (_, contract_id) = test_builder.setup_contract(vec![]);
```

Comment:
See my other posts about updating the test builder to accept initial balances for contracts.

Reply:
I've updated this and the other `setup_contract` references to now initialize with a balance